--- ferm	2015-09-04 12:02:07.843105402 +0200
+++ ferm	2015-09-04 11:58:25.159110930 +0200
@@ -98,6 +98,7 @@
 case "$1" in
     start)
         log_daemon_msg "Starting $DESC" "$NAME"
+        [ -d /etc/ferm/hooks/pre-start.d ] && /bin/run-parts /etc/ferm/hooks/pre-start.d || true
         if configure_ferm; then
 		log_end_msg $?
 	else
@@ -107,17 +108,22 @@
 			log_warning_msg "Looks like the ip_tables module is not loaded, see /etc/modules"
 		fi
 	fi
+        [ -d /etc/ferm/hooks/post-start.d ] && /bin/run-parts /etc/ferm/hooks/post-start.d || true
         ;;
     stop)
         log_daemon_msg "Stopping $DESC" "$NAME"
+        [ -d /etc/ferm/hooks/pre-stop.d ] && /bin/run-parts /etc/ferm/hooks/pre-stop.d || true
         OPTIONS="$OPTIONS --flush"
         configure_ferm stop
         log_end_msg $?
+        [ -d /etc/ferm/hooks/post-stop.d ] && /bin/run-parts /etc/ferm/hooks/post-stop.d || true
         ;;
     reload|restart|force-reload)
         log_begin_msg "Reloading $DESC configuration..."
+        [ -d /etc/ferm/hooks/pre-reload.d ] && /bin/run-parts /etc/ferm/hooks/pre-reload.d || true
         configure_ferm
         log_end_msg $?
+        [ -d /etc/ferm/hooks/post-reload.d ] && /bin/run-parts /etc/ferm/hooks/post-reload.d || true
         ;;
     *)
         N=/etc/init.d/$NAME
