--- Mailman/HTMLFormatter.py
+++ Mailman/HTMLFormatter.py
@@ -56,6 +56,10 @@
                          _('%(realname)s administrative interface')),
                     _(' (requires authorization)'),
                     '<br>',
+                    Link(self.GetScriptURL('admindb'),
+                         _('%(realname)s moderation interface')),
+                    _(' (requires authorization)'),
+                    '<br>',
                     Link(Utils.ScriptURL('listinfo'),
                          _('Overview of all %(hostname)s mailing lists')),
                     '<p>', MailmanLogo()))).Format()
